Tests/UIKitTests/UIViewControllerExtensionsTests.swift

```diff
@@ -120,5 +120,73 @@ final class UIViewControllerExtensionsTests: XCTestCase {
         XCTAssertNil(childViewController.parent)
     }
 
+    #if os(iOS)
+    func testPresentPopover() {
+        let popover = UIViewController()
+        let presentingViewController = UIViewController()
+
+        let window = UIWindow()
+        window.rootViewController = presentingViewController
+        window.rootViewController = presentingViewController
+        window.addSubview(presentingViewController.view)
+        RunLoop.current.run(until: Date())
```

Comment:
Is this your way of waiting for something to happen asynchronously? Generally you would use `XCTestExpectation`, which allows you to wait for callback methods. Maybe this is what it's doing internally, I'm not sure, but it may help to make it clearer what are your intentions.

Reply:
Unfortunately I cannot use the callback because the completion is not called in unit tests due to disabled animations. Also the window + runloop thing is to make sure the presentation is as similar to a real app use case as it can. I found that this is the only way for the `presentedViewController` and `presentingViewController` properties to be set.
I'll write a comment to make it clear

Reply:
What I mean is calling `DispatchQueue.main.async { ... }`. That's my usual trick for letting the run loop perform whatever it needs to. I didn't know about this trick you're using here. I figured that the main thread would be blocked otherwise and you couldn't get the rest of the UI to run without breaking away from it and coming back later. I really like your way!

Reply:
Thanks! Also I try to avoid blocks when I can use another solutions, consecutive lines of code without additional indentation looks "cleaner"

Reply:
Related: when running all the tests some day ago I found that another test gives a warning (not failing though) for a similar reason (UI not properly set up), but I can't remember which one. I'll give it a look